### PR TITLE
Wishlist screen with functionality

### PR DESCRIPTION
**Pull request by dev-enforced**
This PR contains the screens related to the wishlist section of Eleganz Stores Web App that allows users to get started with adding their favourite products available onto their wishlist and from the wishlist they can add their items into the cart.
In the cart also the user can add some items to the wishlist which he likes and remove it they want it to be and the count of items will be displayed on the navbar.
Adding and Removing Items to Wishlist and Cart will occur when the user is logged in. If they aren't they will be redirected to the Login or Signup route to get logged in first

### CODE REVIEW

File: src/components/CartItemList/CartItemList.jsx [hunk 1]
```diff
@@ -30,13 +30,13 @@ const CartItemList = () => {
                                         <button onClick={() => {
                                             quantityUpdate(authState, authDispatch, everyCartItem, "decrement")
                                         }} className="py-2 px-3 cart-item-quantity-btn">
-                                            <FaMinus />
+                                            <Decrease />
```
Comment (Manju-sree) [inline src/components/CartItemList/CartItemList.jsx:33]: It's new thing learnt using react-icons as constants

File: src/components/ProductCard/ProductCard.jsx [hunk 2]
```diff
@@ -57,7 +58,25 @@ const ProductCard = ({ productDetails }) => {
...
+                            </Link>
+                            <button className="btn btn-primary btn-primary-hover gentle-flex-gap flex-align-center" onClick={() => {
+                                removeItemFromWishlist(productDetails, authDispatch, authState);
+                            }}>REMOVE FROM WISHLIST</button>
+                        </> :
+                        <button className="btn btn-primary btn-primary-hover gentle-flex-gap flex-align-center" onClick={() => {
+                            if (signinStatus) {
+                                addItemToWishlist(productDetails, authDispatch, authState);
+                            } else {
+                                navigate("/signin");
```
Comment (Manju-sree) [inline src/components/ProductCard/ProductCard.jsx:75]: good to see you have used concept : Navigate Programmatically - useNavigate

File: src/pages/Wishlist/EmptyWishlist.jsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
```
Comment (Manju-sree) [inline src/pages/Wishlist/EmptyWishlist.jsx:1]: good to see you are maintained separate file for emptywishlist.

**Review by Manju-sree — APPROVED** [review]
I like the way you have written proper clean code which is understood easily by anyone and i liked the way you are re-naming react-icons as per your needs. everything looks fine in this PR. suggestion: add favicon, it would be great if you make card sizes to medium size. it's huge in size now.

File: src/constants/icons.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+import { FaMinus as Decrease, FaPlus as Increase, FaTrash as Remove, FaChevronRight as RightIndicator } from "react-icons/fa";
+import { GoSearch as Search, GoHeart as WishlistIcon } from "react-icons/go";
+import { IoBagOutline as CartIcon, IoCartOutline as CartOutline, IoHeartOutline as HeartOutline } from "react-icons/io5";
+export { Decrease, Increase, Remove, RightIndicator, Search, WishlistIcon, CartIcon, CartOutline, HeartOutline };
```
Comment (sushil-mod) [inline src/constants/icons.js:4]: I liked how you are using different file for icons

File: src/services/Wishlist/removeItemFromWishlist.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+                wishlist: wishlistUpdated
+            }
+        } = await axios.delete(`/api/user/wishlist/${product._id}`, {
+            headers: {
+                authorization: authStateGiven.authenticationToken
+            }
+        })
+        authDispatchFunction({ type: "WISHLIST", payload: wishlistUpdated })
+    } catch (error) {
+        console.log(error);
```
Comment (sushil-mod) [inline src/services/Wishlist/removeItemFromWishlist.js:15]: you can remove this log

**Review by sushil-mod — APPROVED** [review]
just a comment , I liked how you used icons .UI looks awesome . overall good code and also a clean reducer code . nice  work

File: src/services/Wishlist/removeItemFromWishlist.js (hunk 5 above)
Comment (dev-enforced) [reply src/services/Wishlist/removeItemFromWishlist.js:15]: Used for error handling so cannot remove it!!

**Pull request merged**